api/v1alpha1/aws_node_types.go

```diff
@@ -50,6 +54,10 @@ type AWSGenericNodeSpec struct {
 	// PlacementGroup specifies the placement group in which to launch the instance.
 	// +kubebuilder:validation:Optional
 	PlacementGroup *PlacementGroup `json:"placementGroup,omitempty"`
+
+	// Configuration options for the root storage volume.
+	// +kubebuilder:validation:Optional
+	RootVolume *Volume `json:"rootVolume,omitempty"`
```

Comment:
Could we add non-root volumes too please? We've had conversations previously about adding separate disk for image storage, logs, etc to prevent filling up of root disk and killing nodes.

How about something like:

```suggestion
	Volumes *AWSVolumes `json:"volumes,omitempty"`
```

And `AWSVolumes` defined as:

```
type AWSVolumes struct {
	// Configuration options for the root storage volume.
	// +kubebuilder:validation:Required
	Root Volume `json:"root"`

	// Configuration options for non-root storage volumes.
	// +kubebuilder:validation:Optional
	NonRoot []Volume `json:"nonroot,omitempty"`
}
```

Happy to leave the non-root configurations to a follow up PR, but would like to move root volume under `volumes.root` anyway for extensibility. WDYT?

Reply:
Great idea, I'll make this change.

Reply:
Agreed @faiq, I left it optional and let CAPA do its defaults.